data/index.js

```diff
@@ -77,6 +82,16 @@ export function registerSelectors( reducerKey, newSelectors ) {
  * @return {Function} Renders the wrapped component and passes it data.
  */
 export const query = ( mapSelectorsToProps ) => ( WrappedComponent ) => {
+	const store = {
+		getState() {
+			return mapValues( stores, subStore => subStore.getState() );
```

Comment:
Unsure if there's any consequences, but as implemented, it's guaranteed `store.getState() !== store.getState()` in all cases.

Reply:
I think it is only important what `subStore.getState()` returns because we are always referencing `state[ key ]`. See https://github.com/WordPress/gutenberg/pull/4823/files#diff-c4757f613a3f7ef6e0c2a672fdfc5bd4R104.

Reply:
My thinking was that an action triggered means at least one of the sub-store objects will change, which makes it inevitable to update the global store object (`store.getState()`).So any caching mechanism is useless.

Reply:
Right, I didn't process it properly. `query` is applied only once, when a module is imported. It holds the reference to `stores` inside the instance of local `store`.  If you combine with other part of my comment, we are all good 👍